server/portal/apps/onboarding/steps/abstract.py

```diff
@@ -18,6 +19,15 @@ def __init__(self, user):
         self.last_event = None
         self.events = []
 
+        try:
```

Comment:
would it be worth refactoring this to something like (not-tested; for illustration) the following to avoid just catching the broad Exception: 
```
        steps = settings.PORTAL_USER_ACCOUNT_SETUP_STEPS
        step_dict = next(
            step for step in steps if step['step'] == self.step_name(),
            None
        )
       if step_dict:
           try:
                self.settings = step_dict['settings']
           except KeyError:
                self.settings = None
```

Reply:
There's key error and an iteration error. There's also configuration errors which could result in a lot of different types of exceptions. Are you okay leaving it unbounded?